pyuvsim/simsetup.py

```diff
@@ -836,7 +836,77 @@ def _sky_select_calc_rise_set(sky, source_params, telescope_lat_deg=None):
     return sky
 
 
-def initialize_catalog_from_params(obs_params, input_uv=None, return_recarray=True):
+def _read_pyradiosky_file(filename, source_params):
+    """
+    Read in a pyradiosky catalog file.
+
+    Parameters
+    ----------
+    filename : str
+        File to read in.
+    source_params : dict
+        Dictionary specifying options to pyradiosky read methods.
+
+    Returns:
+    --------
+    :class:`pyradiosky.SkyModel`
+        SkyModel object based on filename
+
+    """
+    sky = pyradiosky.SkyModel()
+
+    if filename.endswith("txt"):
+        sky.read_text_catalog(filename)
+    elif filename.endswith('vot'):
+        if 'gleam' in filename:
+            if "spectral_type" in source_params:
+                spectral_type = source_params["spectral_type"]
+            else:
+                warnings.warn("No spectral_type specified for GLEAM, using 'flat'.")
+                spectral_type = "flat"
+            sky.read_gleam_catalog(
+                filename, spectral_type=spectral_type
+            )
+        else:
+            vo_params = {}
+            required_params = ["table_name", "id_column", "flux_columns"]
+            warn_params = {"ra_column": "RAJ2000", "dec_column": "DEJ2000"}
+            for param in required_params:
+                if param not in source_params:
+                    raise ValueError(f"No {param} specified for {filename}.")
+                vo_params[param] = source_params[param]
+            for param, default in warn_params.items():
+                if param not in source_params:
+                    warnings.warn(
+                        f"No {param} specified for {filename}, using default: {default}."
+                    )
+                    vo_params[param] = default
+                else:
+                    vo_params[param] = source_params[param]
+
+            vo_params["reference_frequency"] = None
+            sky.read_votable_catalog(filename, **vo_params)
+    elif filename.endswith('skyh5'):
+        sky.read_skyh5(filename)
```

Comment:
In the future, it'd be good to have a `is_skyh5` function in pyradiosky that could tell you whether it was skyh5 for any input file. Because really you could have any extension and it still be valid.

Reply:
The issue already exists! https://github.com/RadioAstronomySoftwareGroup/pyradiosky/issues/121